src/lib/cert/x509/key_constraint.cpp

```diff
@@ -54,11 +25,11 @@ Key_Constraints find_constraints(const Public_Key& pub_key,
       constraints |= KEY_AGREEMENT;
 
    if(name == "RSA" || name == "ElGamal")
-      constraints |= KEY_ENCIPHERMENT | DATA_ENCIPHERMENT;
+      constraints |= KEY_ENCIPHERMENT | DATA_ENCIPHERMENT | ENCIPHER_ONLY | DECIPHER_ONLY;
 
    if(name == "RSA" || name == "RW" || name == "NR" ||
-      name == "DSA" || name == "ECDSA")
-      constraints |= DIGITAL_SIGNATURE | NON_REPUDIATION;
+      name == "DSA" || name == "ECDSA" || name == "ECGDSA" || name == "ECKCDSA")
+      constraints |= DIGITAL_SIGNATURE | NON_REPUDIATION | KEY_CERT_SIGN | CRL_SIGN;
```

Comment:
@cordney That change works for me. If no constraints are set, the Key Usage extension should be omitted entirely during encoding. Can I suggest to make the operation more explicitly named (eg `verify_cert_constraints_valid_for_key_type`). Both `find_constraints` + `match_constraints` are not really clear about what they do.